### PR TITLE
anki-bin: 2.1.49 -> 2.1.51

### PR DESCRIPTION
**Pull request by dnaq**
anki-bin -> 2.1.51

###### Description of changes
Since the previous version of anki no longer works on nixos-unstable
due to glibc changes the version was bumped to 2.1.51 which works. See #167765

NOTE: The darwin version has not been tested due to lack of a darwin
machine to try it on. The hash and download url has been updated, but
this package should be tested before this pull request is accepted.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by yayayayaka — APPROVED** [review]
Please rename the PR title to `anki-bin: 2.1.49 -> 2.1.50`

Tested on x86_64-linux on the latest nixos-unstable and it fixed the problem, thanks! :tada:

File: pkgs/games/anki/bin.nix [hunk 1]
```diff
@@ -46,9 +47,10 @@ in
 
 if stdenv.isLinux then buildFHSUserEnv (appimageTools.defaultFhsEnvArgs // {
   name = "anki";
+  targetPkgs = pkgs: (with pkgs; [ xorg.libxkbfile krb5 ]);
```
Comment (Atemu) [inline pkgs/games/anki/bin.nix:50]: Why are these needed? Add a comment please.

File: pkgs/games/anki/bin.nix [hunk 2]
```diff
@@ -46,9 +47,10 @@ in
...
 
   runScript = writeShellScript "anki-wrapper.sh" ''
-    exec ${unpacked}/bin/anki
+    DISABLE_QT5_COMPAT=1 exec ${unpacked}/bin/anki
```
Comment (Atemu) [inline pkgs/games/anki/bin.nix:53]: Same here. What does this env var mean, why do we need it and when can we remove it?

**Review by Atemu — CHANGES_REQUESTED** [review]
Result of `nixpkgs-review pr 168300` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>anki-bin</li>
  </ul>
</details>

Also works on x86_64-darwin (rosetta).  
Upstream has an aarch64 build now. Adding that is not necessary for the scope of this PR IMHO but it wouldn't be out of place either.

Please add comments for your additions.

Could you also add `Fixes https://github.com/NixOS/nixpkgs/issues/167765` to the commit message?

File: pkgs/games/anki/bin.nix (hunk 1 above)
Comment (dnaq) [reply pkgs/games/anki/bin.nix:50]: The new anki build refused to start without them, complaining about missing libraries. krb5 was needed for gssapi.

File: pkgs/games/anki/bin.nix (hunk 2 above)
Comment (dnaq) [reply pkgs/games/anki/bin.nix:53]: Anki worked without this environment variable, but printed out a warning to the console saying that it ran in QT5 compatibility mode, and that this environment variable needed to be set to not run it in compatibility mode.

Comment (Atemu) [reply pkgs/games/anki/bin.nix:53]: I think we might not want to override this behaviour. I don't use KDE but it being QT6 might not interface with the rest of the NixOS desktop which is still QT5 AFAIK.

File: pkgs/games/anki/bin.nix (hunk 1 above)
Comment (Atemu) [reply pkgs/games/anki/bin.nix:50]: Great, just add the explanations to the code

File: pkgs/games/anki/bin.nix (hunk 2 above)
Comment (dnaq) [reply pkgs/games/anki/bin.nix]: Addressed all of the issue comments and tried adding support for aarch64 as well, but again, I cannot test that since I don't have access to apple silicon to run anki on. Would be great if someone else could confirm those changes.

**Comment by dnaq** [comment]
Addressed all of the comments and added support for aarch64. Again, that support is untested since I don't have access to apple silicon to test it on. Would be great if someone else could test.

File: pkgs/games/anki/bin.nix [hunk 3]
```diff
@@ -47,6 +54,9 @@ in
 if stdenv.isLinux then buildFHSUserEnv (appimageTools.defaultFhsEnvArgs // {
   name = "anki";
 
+  # Dependencies of anki that aren't part of the appimage
```
Comment (Atemu) [inline pkgs/games/anki/bin.nix:57]: Nit: Anki's Linux package isn't an AppImage.

**Review by Atemu — CHANGES_REQUESTED** [review]
Please update platforms:

```
    platforms = [ "x86_64-linux" "x86_64-darwin" "aarch64-darwin" ];
```

**Comment by TheBlackBeans** [comment]
I think, due to add-ons compatibility reasons, it should be possible to choose between Qt5 and Qt6.

**Comment by robert-elles** [comment]
Hello, 
trying this new version anki doesn't start and I get the following error:

Anki starting...
Initial setup...
Running with temporary Qt5 compatibility shims.
Run with DISABLE_QT5_COMPAT=1 to confirm compatibility with Qt6.
Preparing to run...
libGL error: MESA-LOADER: failed to open swrast: /usr/lib/libstdc++.so.6: version `GLIBCXX_3.4.29' not found (required by /run/opengl-driver/lib/dri/swrast_dri.so) (search paths /run/opengl-driver/lib/dri, suffix _dri)
libGL error: failed to load driver: swrast
Command '['gsettings', 'get', 'org.gnome.desktop.interface', 'gtk-theme']' returned non-zero exit status 1.
Qt warning: WebEngineContext used before QtWebEngineCore::initialize() or OpenGL context creation failed. 
Qt warning: QGLXContext: Failed to create dummy context 
/nix/store/xwi84ffanr87553zjway3c6ygpp8hs3b-anki-bin-2.1.50/share/anki/lib/PyQt6/Qt6/libexec/QtWebEngineProcess: /usr/lib/libz.so.1: no version information available (required by /nix/store/xwi84ffanr87553zjway3c6ygpp8hs3b-anki-bin-2.1.50/share/anki/lib/PyQt6/Qt6/libexec/../lib/libQt6WebEngineCore.so.6)
/nix/store/xwi84ffanr87553zjway3c6ygpp8hs3b-anki-bin-2.1.50/share/anki/lib/PyQt6/Qt6/libexec/QtWebEngineProcess: /usr/lib/libz.so.1: no version information available (required by /nix/store/xwi84ffanr87553zjway3c6ygpp8hs3b-anki-bin-2.1.50/share/anki/lib/PyQt6/Qt6/libexec/../lib/libQt6Gui.so.6)
/nix/store/xwi84ffanr87553zjway3c6ygpp8hs3b-anki-bin-2.1.50/share/anki/lib/PyQt6/Qt6/libexec/QtWebEngineProcess: /usr/lib/libz.so.1: no version information available (required by /nix/store/xwi84ffanr87553zjway3c6ygpp8hs3b-anki-bin-2.1.50/share/anki/lib/PyQt6/Qt6/libexec/../lib/libQt6Gui.so.6)
/nix/store/xwi84ffanr87553zjway3c6ygpp8hs3b-anki-bin-2.1.50/share/anki/lib/PyQt6/Qt6/libexec/QtWebEngineProcess: /usr/lib/libz.so.1: no version information available (required by /nix/store/xwi84ffanr87553zjway3c6ygpp8hs3b-anki-bin-2.1.50/share/anki/lib/PyQt6/Qt6/libexec/../lib/libQt6WebEngineCore.so.6)
/nix/store/xwi84ffanr87553zjway3c6ygpp8hs3b-anki-bin-2.1.50/share/anki/lib/PyQt6/Qt6/libexec/QtWebEngineProcess: /usr/lib/libz.so.1: no version information available (required by /nix/store/xwi84ffanr87553zjway3c6ygpp8hs3b-anki-bin-2.1.50/share/anki/lib/PyQt6/Qt6/libexec/../lib/libQt6Gui.so.6)
/nix/store/xwi84ffanr87553zjway3c6ygpp8hs3b-anki-bin-2.1.50/share/anki/lib/PyQt6/Qt6/libexec/QtWebEngineProcess: /usr/lib/libz.so.1: no version information available (required by /nix/store/xwi84ffanr87553zjway3c6ygpp8hs3b-anki-bin-2.1.50/share/anki/lib/PyQt6/Qt6/libexec/../lib/libQt6Gui.so.6)
Qt info: 

GLImplementation: 
Surface Type: OpenGL
Surface Profile: NoProfile
Surface Version: 2.0
Using Default SG Backend: yes
Using Software Dynamic GL: no
Using Angle: no

Init Parameters:
  *  application-name Anki 
  *  browser-subprocess-path /nix/store/xwi84ffanr87553zjway3c6ygpp8hs3b-anki-bin-2.1.50/share/anki/lib/PyQt6/Qt6/libexec/QtWebEngineProcess 
  *  disable-features DnsOverHttpsUpgrade,ConsolidatedMovementXY,InstalledApp,BackgroundFetch,WebOTP,WebPayments,WebUSB,PictureInPicture 
  *  disable-gpu  
  *  disable-setuid-sandbox  
  *  disable-speech-api  
  *  enable-features NetworkServiceInProcess,TracingServiceInProcess 
  *  enable-threaded-compositing  
 
Starting main loop...

**Review by Atemu — CHANGES_REQUESTED** [review]
I don't use any but if add-on compatibility is a concern, we should either allow the choice of both QT versions or only offer QT5 (where available).

**Review by yayayayaka — APPROVED** [review]
Tested on x86_64-linux and can confirm it's working again.

**Comment by euank** [comment]
Thanks for doing this update!

The commit message subject should also be formatted as `anki-bin: 2.1.49 -> 2.1.51`

**Comment by Patryk27** [comment]
I can also confirm this works, thanks! 🙂

**Comment by dnaq** [comment]
Anki hasn’t been working at all in unstable for a while now. If you merge
this pull request Anki will work for most people, then new issues can be
made for whatever reason needed.

**Comment by SuperSandro2000** [comment]
What pull request? Also instead of creating issues in our big collection we should try to fix issues right away.

**Comment by dnaq** [comment]
I think GitHub is acting up right now, that message was written by me five months ago, no idea why it says two hours
